wallet-new/src/Cardano/Wallet/Kernel/WalletException.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+module Cardano.Wallet.Kernel.WalletException where
+
+import           Universum
+
+import           Data.Aeson (ToJSON, toJSON)
+import           Data.Typeable (cast)
+import           Formatting.Buildable (build)
+import           Prelude (show)
+
+import           Cardano.Wallet.API.V1.Errors (ToHttpErrorStatus (..),
+                     ToServantError (..))
+
+data WalletException = forall e . (ToHttpErrorStatus e, Exception e, Buildable e) => WalletException e
+
+-- FIXME: Generics somewhow?
+
+instance Show WalletException where
+    show (WalletException e) = Prelude.show e
+
+instance ToJSON WalletException where
+    toJSON (WalletException e) = toJSON e
+
+instance ToHttpErrorStatus WalletException where
+    toHttpErrorStatus (WalletException e) = toHttpErrorStatus e
+
+instance ToServantError WalletException where
+    declareServantError (WalletException e) = declareServantError e
+
+instance Buildable WalletException where
+    build (WalletException e) = build e
+
+instance Exception WalletException
```

Comment:
I don't think we need all those instances. Only `Show` is required to define the `Exception`.  So it could basically be:

```hs
data WalletException = forall e.
    ( ToHttpErrorStatus e
    , HasDiagnostic e
    , Exception e
    ) => WalletException { getWalletException :: e }

instance Prelude.Show WalletException where
    show (WalletException e) = Prelude.show e

instance Exception WalletException
```